From c54df508da76f989dadea38adb9001481a9e53b8 Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@gmail.com>
Date: Mon, 29 Jan 2024 12:19:38 -0800
Subject: [PATCH 6/7] rocm 6.0.0 HIP_PLATFORM_AMD fix

- ROCM 6.0.0 uses HIP_PLATFORM_AMD instead
of deprecated HIP_PLATFORM_HCC with pytorch.
- fixes following build error:
-c /home/lamikr/own/rocm/src/sdk/rocm_sdk_builder_600/src_projects/pytorch/aten/src/ATen/hip/nvrtc_stub/ATenNVRTC.cpp
In file included from /home/lamikr/own/rocm/src/sdk/rocm_sdk_builder_600/src_projects/pytorch/aten/src/ATen/hip/ATenHIPGeneral.h:4,
                 from /home/lamikr/own/rocm/src/sdk/rocm_sdk_builder_600/src_projects/pytorch/aten/src/ATen/hip/nvrtc_stub/ATenNVRTC.h:4,
                 from /home/lamikr/own/rocm/src/sdk/rocm_sdk_builder_600/src_projects/pytorch/aten/src/ATen/hip/nvrtc_stub/ATenNVRTC.cpp:2:
/opt/rocm_sdk_600/include/hip/hip_runtime.h:66:2: error: #error ("Must define exactly one of __HIP_PLATFORM_AMD__ or __HIP_PLATFORM_NVIDIA__");
   66 | #error("Must define exactly one of __HIP_PLATFORM_AMD__ or __HIP_PLATFORM_NVIDIA__");
      |  ^~~~~
In file included from /opt/rocm_sdk_600/include/hip/hip_runtime.h:70:
/opt/rocm_sdk_600/include/hip/hip_runtime_api.h:8575:2: error: #error ("Must define exactly one of __HIP_PLATFORM_AMD__ or __HIP_PLATFORM_NVIDIA__");
 8575 | #error("Must define exactly one of __HIP_PLATFORM_AMD__ or __HIP_PLATFORM_NVIDIA__")

Signed-off-by: Mika Laitio <lamikr@gmail.com>
---
 caffe2/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/caffe2/CMakeLists.txt b/caffe2/CMakeLists.txt
index 748363725bc..d91f3df40fe 100644
--- a/caffe2/CMakeLists.txt
+++ b/caffe2/CMakeLists.txt
@@ -694,7 +694,7 @@ if(USE_ROCM)
   add_library(caffe2_nvrtc SHARED ${ATen_NVRTC_STUB_SRCS})
   target_link_libraries(caffe2_nvrtc ${PYTORCH_HIP_LIBRARIES} ${ROCM_HIPRTC_LIB})
   target_include_directories(caffe2_nvrtc PRIVATE ${CMAKE_BINARY_DIR})
-  target_compile_definitions(caffe2_nvrtc PRIVATE USE_ROCM __HIP_PLATFORM_HCC__)
+  target_compile_definitions(caffe2_nvrtc PRIVATE USE_ROCM __HIP_PLATFORM_AMD__)
   install(TARGETS caffe2_nvrtc DESTINATION "${TORCH_INSTALL_LIB_DIR}")
 endif()
 
-- 
2.41.0

